pkg/test/framework/config.go

```diff
@@ -112,6 +115,40 @@ func (c *configManager) DeleteYAMLOrFail(t test.Failer, ns string, yamlText ...s
 	}
 }
 
+func (c *configManager) WaitForConfig(ctx resource.Context, ns string, yamlText ...string) error {
+	var outErr error
+	for _, c := range c.ctx.Clusters() {
+		ik, err := istioctl.New(ctx, istioctl.Config{Cluster: c})
+		if err != nil {
+			return err
+		}
+
+		for _, config := range yamlText {
+			config := config
+
+			// TODO(https://github.com/istio/istio/issues/37324): It's currently unsafe
```

Comment:
this sounds like we could be in for surprises?  How big of a concern is this issue?

Reply:
I'm not entirely sure TBH. I just realized it was an issue and raised https://github.com/istio/istio/pull/37325 as a potential mitigation. I suspect fixing istioctl would be a major undertaking and is unlikely to ever happen.